### PR TITLE
Fix register features

### PR DESCRIPTION
**Pull request by tarockey**
- For issues:
    - #84 
    - #86 
- Updates FeathrClient and _feature_registry to appropriate raise RuntimeErrors
- Updates _feature_registry to handle multiple .conf files in a single folder (as generated by build_features)
- Formatted templated .conf files to be more human readable
- Small formatting adjustments

### CODE REVIEW

**Comment by xiaoyongzhu** [comment]
Thanks for the PR! The test is failed because credentials are not spread to forked repo (will fix that hopefully soon), and others are all good to me. I've approved the changes and wait for @hangfei and @jaymo001 for their input on the string format changes.

File: feathr_project/feathr/_feature_registry.py [hunk 1]
```diff
@@ -645,69 +649,84 @@ def save_to_feature_config(self, workspace_path: Path):
...
-                            {% endif %}
-                        {% endfor %}
-                    }
-                    """)
-        
-        request_feature_configs = tm.render(feature_anchors=repo_definitions.feature_anchors)
+        config_file_name = "feature_conf/auto_generated_request_features.conf"
+        tm = Template(
+            """
+// THIS FILE IS AUTO GENERATED. PLEASE DO NOT EDIT.
```
Comment (xiaoyongzhu) [inline feathr_project/feathr/_feature_registry.py:666]: @jaymo001 @hangfei Are you OK with this?

**Review by xiaoyongzhu — DISMISSED** [review]

File: feathr_project/feathr/_feature_registry.py (hunk 1 above)
Comment (jaymo001) [reply feathr_project/feathr/_feature_registry.py:666]: LGTM. Thanks a lot for the contribution.

File: feathr_project/feathr/_feature_registry.py [hunk 2]
```diff
@@ -791,12 +813,15 @@ def get_features_from_registry(self, project_name: str, workspace_path: str):
...
-        logger.info("Writing feature configuration from feathr registry to {}", feature_conf_file)
-        
+        logger.info(
+            "Writing feature configuration from feathr registry to {}", feature_conf_file)
+
         with open(feature_join_file, "w") as offline_config:
             offline_config.write(feature_join_conf_content)
-        logger.info("Writing offline configuration from feathr registry to {}", feature_join_file)
+        logger.info(
+            "Writing offline configuration from feathr registry to {}", feature_join_file)
```
Comment (jaymo001) [inline feathr_project/feathr/_feature_registry.py:822]: Is the indentation expected? They seem to be inconsistent with other code. Did you do this manually, or with some plugin/automatically.

File: feathr_project/feathr/_feature_registry.py (hunk 1 above)
Comment (tarockey) [reply feathr_project/feathr/_feature_registry.py:666]: and for reference - the only change here was to adjust the indentation of the block quote - it was causing weird formatting in the generated .conf files that made them difficult to read as a person - this is certainly not a mission critical change

**Comment by tarockey** [comment]
> 

No problem! Should I try and merge into a different branch so that the tests can be run before merging into main?

File: feathr_project/feathr/_feature_registry.py (hunk 2 above)
Comment (tarockey) [reply feathr_project/feathr/_feature_registry.py:822]: Ah, must just be the formatter running in my IDE - I can revert that

**Comment by tarockey** [comment]
> Thanks for the PR! The test is failed because credentials are not spread to forked repo (will fix that hopefully soon), and others are all good to me. I've approved the changes and wait for @hangfei and @jaymo001 for their input on the string format changes.

No problem! Should I try and merge into a different branch so that the tests can be run before merging into main?

File: feathr_project/feathr/_feature_registry.py (hunk 2 above)
Comment (tarockey) [reply feathr_project/feathr/_feature_registry.py]: Updated my PR to match existing formatting in the repo

**Review by jaymo001 — APPROVED** [review]
LGTM. Great work!

**Review by xiaoyongzhu — APPROVED** [review]

**Comment by xiaoyongzhu** [comment]
@tarockey Thanks for the PR! I just merged it.

**Pull request merged**